### PR TITLE
MM-9787: Transition [lr]hs open state to redux, fixing Safari race condition.

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Transition the open/closed state of the lhs/rhs/rhs menu to Redux, with partial changes to do the same for WebRTC. This addresses a previous race condition around toggling the various `move--left`, `move--right`, `move--left-small`, etc. classes on all the various moving elements, allowing Safari to consistently open the menu when walking through the tutorial.

Note that the WebRTC code uses the old Flux pattern, and I've made the minimal possible changes to keep it working without diving into a rathole to make it any better. This beta feature is "scheduled" to be converted to a plugin, and isn't worth moving to pure Redux right now.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9787

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

**Comment by jespino** [comment]
Looks very good, but it is a huge PR. I didn't made a deep code review, so I going to test it manually a bit too.

File: actions/views/lhs.js [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+    type: ActionTypes.OPEN_LHS,
+});
+
+export const close = () => (dispatch) => dispatch({
+    type: ActionTypes.CLOSE_LHS,
+});
+
 export async function initTeamChangeActions(teamId) {
-    await fetchMyChannelsAndMembers(teamId)(dispatch, getState);
+    await fetchMyChannelsAndMembers(teamId)(store.dispatch, getState);
```
Comment (enahum) [inline actions/views/lhs.js]: `store.getState` ? also you could do `await store.dispatch(fetchMyChannelsAndMembers(teamId));`

Comment (lieut-data) [reply actions/views/lhs.js]: Gah, you keep suggesting this and I keep forgetting to integrate it into my PRs. Thank you -- I'll clean this up :)

Comment (lieut-data) [reply actions/views/lhs.js]: As it turns out, this method is entirely unused anyway. I'll push up its removal with any changes required after feedback from @jespino.

**Comment by lieut-data** [comment]
@enahum, changes pushed. Let me know if you have any more concerns :)

**Pull request merged**